core/Azure.Mcp.Core/src/Areas/Server/Commands/ToolLoading/NamespaceToolLoader.cs

```diff
@@ -395,6 +383,7 @@ private async Task<CallToolResult> InvokeChildToolAsync(
             }
 
             var currentActivity = Activity.Current;
+            currentActivity?.SetTag(TagName.ToolId, cmd.Id);
```

Comment:
Suggestion is to put this on L406 because a tag related is also set there so its all grouped together